### DISCUSSION TITLE
how can `bunFile.size` be immediate?

### DISCUSSION THREAD

**Discussion by yukulele** [oven-sh/bun#6140] [Q&A]
considering this code:
```ts
const floppyFileSize = new file('/dev/fd0/info.txt').size
```

I was expecting to have an `await` somewhere, but no.

How `bunFile.size` can get the file size immediately?
Is it a synchronous (blocking) call?

**Comment by melroy89** [comment]
Yes it's a blocking call. See https://bun.sh/docs/api/file-io

**Reply by melroy89** [reply]
But using .text() of .stream() is not a blocking call.

**Reply by yukulele** [reply]
Unless I've misread it, the API only says that `.size` returns the number of bytes, but it doesn't explain how it works, nor does it specify that it's blocking.

**Reply by melroy89** [reply]
I suspect without looking at the code. The file size is read from partition information eg inode data under Linux. Without the need to actually read the file itself.

Reading the data from the file is however more i/o intensive thus  asynchrone functions are present. Avoiding to block the main loop under Javascript.
